app/views/dispatch/_modal.html.haml

```diff
@@ -19,14 +22,12 @@
 
             %td{style: 'width: 340px;'}
               #conversation-messages
-
-              - if @conversation
-                .msg-reply
-                  #msg-input{ contenteditable: true}
-                  = button_tag 'Send', id: 'msg-submit', class: 'btn btn-send btn-primary btn-xs pull-right'
-              - else
-                %p
-                  = 'No number on record for this voter. No messages available.'
+              .msg-reply
+                #msg-input{ contenteditable: true}
+                = button_tag 'Send', id: 'msg-submit', class: 'btn btn-send btn-primary btn-xs pull-right'
+              %p.no-phone
+                No number on record for this voter.
```

Comment:
I'm not sure I follow this - in the case of no voter phone, it seems to now be showing "No number on record" _as well as_ the message-input & "Send" button, where before it showed one or the other.

Reply:
it essentially lazy-loads a lot of the data, when you click a table row, which is why you see the weird javascript-interacting-unexpectedly-with-haml stuff. before the code was trying to show one or the other--but it didn't have sufficient data at that point, so it always showed just 'no phone number'. This is a hack for sure, but now it initially shows both, and once it has the correct data when it loads _messages, it shows or hides the appropriate elements. There's probably a better way to do this even within this hackfest, but this seems to be working so  I'm going to leave it. Will test with a few browsers though.